### PR TITLE
PERF: Block.apply

### PR DESCRIPTION
**Pull request by jbrockmendel**
```
from asv_bench.benchmarks.groupby import *
self = GroupByMethods()
self.setup('object', 'ffill', 'direct', 1)

%timeit self.time_dtype_as_group('object', 'ffill', 'direct', 1)
67.6 µs ± 708 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)  # <- PR
82 µs ± 2.24 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)  # <- master
```

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -373,8 +373,7 @@ def apply(self, func, **kwargs) -> list[Block]:
         apply the function to my values; return a block if we are not
         one
         """
-        with np.errstate(all="ignore"):
```
Comment (jreback) [inline pandas/core/internals/blocks.py:376]: and we don't have warnings?

Comment (jbrockmendel) [reply pandas/core/internals/blocks.py:376]: i see 4 in the tests, all coming from np.sqrt or np.log with negative numbers, which it isn't obvious to me we _should_ suppress

Comment (jreback) [reply pandas/core/internals/blocks.py:376]: that's fine but https://github.com/pandas-dev/pandas/pull/43609/checks?check_run_id=3626217065

need to catch in the test then (which is more explict).

**Comment by jbrockmendel** [comment]
greenish ex circleci

**Pull request merged**